src/types/controlPlaneConfig.ts

```diff
@@ -8,7 +8,27 @@ export type DestinationDefinition = {
   Config: Record<string, unknown>;
 };
 
-export type Destination<DestinationConfig = Record<string, unknown>> = {
+export type AccountDefinition<Config = { refreshOAuthToken?: string }> = {
+  Name: string;
+  Config: Config;
+  AuthenticationType: string;
+};
+export type AccountWithDefinition<
+  OptionsT = Record<string, unknown>,
+  SecretT = Record<string, unknown>,
+  AccountDefinitionT = AccountDefinition,
+> = {
+  ID: string;
+  Options: OptionsT;
+  Secret: SecretT;
+  AccountDefinition: AccountDefinitionT;
+};
+
+export type Destination<
+  DestinationConfig = Record<string, unknown>,
+  DeliveryAccountT = AccountWithDefinition,
+  RegulationAccountT = AccountWithDefinition,
```

Comment:
If we are using deleteAccount in destination object, let's use one consistent wording 'DeleteAccountT'.